SensusService/DataStores/DataStore.cs

```diff
@@ -249,7 +251,7 @@ public virtual void Start()
             }
         }
 
-        public Task AddAsync(Datum datum, CancellationToken cancellationToken)
+        public Task AddAsync(Datum datum, CancellationToken cancellationToken, bool runLocalCommitOnStore)
```

Comment:
We'll always store, so `runLocalCommitOnStore` is a poor name. Also, this object is not necessarily a `LocalDataStore` (it could be a `RemoteDataStore`). So rename to `commit`.